tests/unit/document_tests.py

```diff
@@ -240,6 +242,26 @@ def test_fetch_existing_document_with_docid(self):
         new_doc.fetch()
         self.assertEqual(new_doc, doc)
 
+    def test_appended_error_message_using_save_with_invalid_key(self):
+        """
+        Test that saving a document with an invalid remote key will
+        throw an HTTPError with additional error details from util
+        method append_response_error_content.
+        """
+        # First create the document
+        doc = Document(self.db, 'julia006')
+        # Add an invalid key and try to save document
+        doc['_invalid_key'] = 'jules'
+        resp = None
+        with self.assertRaises(requests.HTTPError) as cm:
+            resp = doc.save()
+        err = cm.exception
+        self.assertEqual(
+            str(err),
+            '500 Server Error: Internal Server Error for url: %s doc_validation '
```

Comment:
I guess this will need updating when the server side error is fixed to be a 4xx. 
Perhaps we should assert that the string ends with the error/reason we expect.
Then maybe have a second assertion of the response code 400? (or 500 until fixed).